src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java

```diff
@@ -123,37 +129,43 @@ public AFCalculationResult getLog10PNonRef(final VariantContext vc, final int de
         // we compute posteriors here and don't have the same prior that AFCalculationResult expects.  Therefore, we
         // give it our posterior as its "likelihood" along with a flat dummy prior
         final double[] dummyFlatPrior = {-1e-10, -1e-10};   //TODO: HACK must be negative for AFCalcResult
-        final double[] log10PosteriorOfNoVariantYesVariant = {log10PNoVariant, Math.log10(1 - Math.pow(10, log10PNoVariant))};
+        final double[] log10PosteriorOfNoVariantYesVariant = {log10PNoVariant, MathUtils.log10OneMinusPow10(log10PNoVariant)};
 
         return new AFCalculationResult(integerAltAlleleCounts, alleles, log10PosteriorOfNoVariantYesVariant, dummyFlatPrior, log10PRefByAllele);
     }
 
+    // effectiveAlleleCounts[allele a] = SUM_{genotypes g} (posterior_probability(g) * num_copies of a in g), which we denote as SUM [n_g p_g]
+    // for numerical stability we will do this in log space:
+    // count = SUM 10^(log (n_g p_g)) = SUM 10^(log n_g + log p_g)
+    // thanks to the log-sum-exp trick this lets us work with log posteriors alone
     private double[] effectiveAlleleCounts(final VariantContext vc, final double[] log10AlleleFrequencies) {
         final int numAlleles = vc.getNAlleles();
         Utils.validateArg(numAlleles == log10AlleleFrequencies.length, "number of alleles inconsistent");
-        final double[] result = new double[numAlleles];
+        final double[] log10Result = new double[numAlleles];
+        Arrays.fill(log10Result, Double.NEGATIVE_INFINITY);
         for (final Genotype g : vc.getGenotypes()) {
             if (!g.hasLikelihoods()) {
                 continue;
             }
             final GenotypeLikelihoodCalculator glCalc = GL_CALCS.getInstance(g.getPloidy(), numAlleles);
-            final double[] genotypePosteriors = normalizedGenotypePosteriors(g, glCalc, log10AlleleFrequencies);
+
+            final double[] log10GenotypePosteriors = log10NormalizedGenotypePosteriors(g, glCalc, log10AlleleFrequencies);
 
             new IndexRange(0, glCalc.genotypeCount()).forEach(genotypeIndex ->
                 glCalc.genotypeAlleleCountsAt(genotypeIndex).forEachAlleleIndexAndCount((alleleIndex, count) ->
-                        result[alleleIndex] += genotypePosteriors[genotypeIndex] * count));
+                        log10Result[alleleIndex] = MathUtils.approximateLog10SumLog10(log10Result[alleleIndex], log10GenotypePosteriors[genotypeIndex] + MathUtils.log10(count))));
```

Comment:
The approximate version is a lot faster and is still very accurate, but the new qual score is so inexpensive compared to calculating PLs that your suggestion will probably be hardly noticeable in the overall runtime.